src/Language/Lambda/Impl.hs

```diff
@@ -118,37 +119,94 @@ instance ZipMatchK TermSig
 -- | Match 'Raw.Ident' via 'Eq'.
 instance ZipMatchK Raw.VarIdent where zipMatchWithK = zipMatchViaEq
 
-instance ZipMatchK Raw.MetaVarIdent where zipMatchWithK = zipMatchViaEq
+instance ZipMatchK Raw.MetavarIdent where zipMatchWithK = zipMatchViaEq
 
 instance ZipMatchK Raw.Type where zipMatchWithK = zipMatchViaEq
 
+-- instance TypedSignature TermSig where
+--   mapSigWithTypes f g sig t = case sig of
+--     AppSig h x -> AppSig (fmap f f) (fmap f x)
+--     Lam typ binder body | Raw.Fun _ b <- t -> Lam typ binder (f body (typ, b))
+--     MetavarSig metavar args -> MetavarSig metavar (fmap (mapSigWithTypes f g) args)
+
+instance TypedSignature TermSig Raw.Type where
+  mapSigWithTypes
+    :: NameMap n Raw.Type
+    -> ( ScopedAST binder (Sum (AnnSig Raw.Type TermSig) ext) n
+         -> (Raw.Type, Raw.Type)
+         -> scopedTerm
+       )
+    -> ( AST binder (Sum (AnnSig Raw.Type TermSig) ext) n
+         -> Raw.Type
+         -> term
+       )
+    -> TermSig
+        (ScopedAST binder (Sum (AnnSig Raw.Type TermSig) ext) n)
+        (AST binder (Sum (AnnSig Raw.Type TermSig) ext) n)
+    -> Raw.Type
+    -> Maybe (TermSig scopedTerm term)
```

Comment:
Do we need the explicit type signature?

Reply:
Probably no, it was automatically suggested by HLS, this is why I applied. If you think it's redundant, it could be removed.